pkg/api/message/service.go

```diff
@@ -452,3 +454,19 @@ func (s *Service) validateClientInfo(clientEnv *envelopes.ClientEnvelope) error
 
 	return nil
 }
+
+func (s *Service) waitForGatewayPublish(stagedEnv queries.StagedOriginatorEnvelope) {
```

Comment:
We should at least pass the context here so we can return early if the context is canceled. Maybe also add a deadline to the context so this doesn't wait forever if something goes wrong?